### PR TITLE
#388 + #418  Port local-key-binding, current-local-map, current-global-map

### PR DESCRIPTION
**Pull request by wizmer**

### CODE REVIEW

**Comment by shaleh** [comment]
rustfmt fail

File: rust_src/src/keymap.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+use remacs_macros::lisp_fn;
+use lisp::{LispObject};
+use remacs_sys::{Flookup_key};
+use threads::ThreadState;
+
+// Return the binding for command KEYS in current local keymap only.
+//   KEYS is a string or vector, a sequence of keystrokes.
```
Comment (shanavas786) [inline rust_src/src/keymap.rs]: remove extra spaces

File: src/keymap.c [hunk 2]
```diff
@@ -1675,24 +1675,6 @@ specified buffer position instead of point are used.
...
-  (Lisp_Object keys, Lisp_Object accept_default)
-{
-  register Lisp_Object map;
-  map = BVAR (current_buffer, keymap);
-  if (NILP (map))
-    return Qnil;
-  return Flookup_key (map, keys, accept_default);
-}
-
-/* GC is possible in this function if it autoloads a keymap.  */
```
Comment (Wilfred) [inline src/keymap.c]: I think this comment applies to the function below, so I'd leave it in.

Comment (wizmer) [reply src/keymap.c]: There was the same comment above and below, so the one above will now be at the same place than this one.

File: rust_src/src/keymap.rs [hunk 3]
```diff
@@ -0,0 +1,41 @@
+use remacs_macros::lisp_fn;
```
Comment (shaleh) [inline rust_src/src/keymap.rs]: You are missing the module level comment. First line should be something like `//! Keymap support`.

File: rust_src/src/keymap.rs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+use lisp::LispObject;
+use remacs_sys::{current_global_map as _current_global_map, Flookup_key};
+use threads::ThreadState;
+
+
+// Return the binding for command KEYS in current local keymap only.
+//   KEYS is a string or vector, a sequence of keystrokes.
```
Comment (shaleh) [inline rust_src/src/keymap.rs]: The function level comment should be 3 slashes, not 2.

File: rust_src/src/keymap.rs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        unsafe {
+            LispObject::from(Flookup_key(
+                map.to_raw(),
+                keys.to_raw(),
+                accept_default.to_raw(),
+            ))
+        }
+    }
+}
+
```
Comment (shaleh) [inline rust_src/src/keymap.rs]: You are inconsistent with line spacing. There should be one line not too.

**Comment by shaleh** [comment]
The fundamentals are there. Now we just need to fine details. Good work so far.

**Comment by wizmer** [comment]
I have amended my PR, the formating should be good now

File: rust_src/src/keymap.rs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+use remacs_sys::{current_global_map as _current_global_map, Flookup_key};
+use threads::ThreadState;
+
+
+/// Return the binding for command KEYS in current local keymap only.
+///   KEYS is a string or vector, a sequence of keystrokes.
+///   The binding is probably a symbol with a function definition.
+///
+///   If optional argument ACCEPT-DEFAULT is non-nil, recognize default
+///   bindings; see the description of `lookup-key' for more details about this.
```
Comment (brotzeit) [inline rust_src/src/keymap.rs]: I think you need to reformat the docstring.

**Comment by Wilfred** [comment]
Perfect! I'm happy, just waiting on Travis to go green.

**Pull request merged**